Abusing CSharp/Code/LinqToOperators/OperatorEnumerable.cs

```diff
@@ -286,34 +286,16 @@ public override int GetHashCode()
 
         private static IEnumerable<object> RepeatSequence(IEnumerable<object> source, int times)
         {
-            for (int i = 0; i < times; i++)
-            {
-                foreach (var item in source)
-                {
-                    yield return item;
-                }
-            }
+            return Enumerable.Range(1,times)
+                .SelectMany(x => source);
         }
 
         private static IEnumerable<object> Shuffle(IEnumerable<object> source)
         {
             // Never do this!
-            Random rng = new Random();
-            var array = source.ToArray();
-            // Note i > 0 to avoid final pointless iteration
-            for (int i = array.Length - 1; i > 0; i--)
-            {
-                // Swap element "i" with a random earlier element it (or itself)
-                int swapIndex = rng.Next(i + 1);
-                object tmp = array[i];
-                array[i] = array[swapIndex];
-                array[swapIndex] = tmp;
-            }
-            // Lazily yield (avoiding aliasing issues etc)
-            foreach (object element in array)
-            {
-                yield return element;
-            }
+            var rng = new Random();
```

Comment:
I'd definitely prefer not to do this one. Not only is this less efficient, but I shy away from ordering by something that's unstable.

Reply:
I did some testing and I see your point on efficiency in large enough item count to measure. However, when I try your version of the function, I don't see the determinism in sorting your statement implies. The existing code appears to just jumble up the items randomly as no seed value is provided. Can you provide some insight into what you meant?

Reply:
It's not a matter of determinism - it's a matter of intent. I view OrderBy as an ordering operation where I should be able to consistently compare any two items. As it happens I suspect that the projection is only applied once per element, but that feels like something the implementation shouldn't have to rely on.

Reply:
Thanks for your time and insights. I enjoyed your abusing C# talk video from youtube and found the metaphors available in this section the most interesting to me, which is why I ended up proposing some changes.